drivers/hmis/app/models/hmis/hud/processors/base.rb

```diff
@@ -300,4 +319,44 @@ def safe_parse_date(string)
   rescue ArgumentError
     raise "unexpected value for date \"#{string}\""
   end
+
+  private def process_files(cded, value)
+    if cded.repeats?
+      Array.wrap(value).map do |val|
+        process_id_to_file(val)
+      end
+    else
+      process_id_to_file(value)
+    end
+  end
+
+  private def process_id_to_file(value)
+    return value unless value
+
+    if value.to_i.to_s == value.to_s
+      # If value is a string representation of an integer, then it should be the ID of an existing File record.
+      # This is a file we've seen and saved before.
+      Hmis::File.find(value)
```

Comment:
Access control check here

Reply:
is this custom data element already associated with this custom assessment (the owner of this processor)? throw an error if not